test/e2e/framework/service/jig.go

```diff
@@ -890,19 +882,24 @@ func testEndpointReachability(endpoint string, port int32, protocol v1.Protocol,
 	cmd := ""
 	switch protocol {
 	case v1.ProtocolTCP:
-		cmd = fmt.Sprintf("nc -zv -t -w 2 %s %v", endpoint, port)
+		cmd = fmt.Sprintf("echo hostName | nc -v -t -w 2 %s %v", endpoint, port)
```

Comment:
Just a question: What is a purpose of `echo hostName` ?

Reply:
the pod returns its hostname https://github.com/tallclair/kubernetes/blob/c05a350b4f16bbf1a709b82a97584efd153e6a8e/test/images/agnhost/README.md#netexec